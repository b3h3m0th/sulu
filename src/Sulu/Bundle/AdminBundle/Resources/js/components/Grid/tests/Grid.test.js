// @flow
import React from 'react';
import {render} from '@testing-library/react';
import Grid from '../Grid';

test('Render a Grid with Items in all sizes', () => {
    const {container} = render(
        <Grid>
            <Grid.Item colSpan={1} />
            <Grid.Item colSpan={2} />
            <Grid.Item colSpan={3} />
            <Grid.Item colSpan={4} />
            <Grid.Item colSpan={5} />
            <Grid.Item colSpan={6} />
            <Grid.Item colSpan={7} />
            <Grid.Item colSpan={8} />
            <Grid.Item colSpan={9} />
            <Grid.Item colSpan={10} />
            <Grid.Item colSpan={11} />
            <Grid.Item colSpan={12} />
        </Grid>
    );
    expect(container).toMatchSnapshot();
});

test('Render a Grid with Sections', () => {
    const {container} = render(
        <Grid>
            <Grid.Section colSpan={4}>
                <Grid.Item colSpan={1} />
                <Grid.Item colSpan={2} />
                <Grid.Item colSpan={3} />
                <Grid.Item colSpan={4} />
                <Grid.Item colSpan={5} />
                <Grid.Item colSpan={6} />
            </Grid.Section>
            <Grid.Section colSpan={8}>
                <Grid.Item colSpan={7} />
                <Grid.Item colSpan={8} />
                <Grid.Item colSpan={9} />
                <Grid.Item colSpan={10} />
                <Grid.Item colSpan={11} />
                <Grid.Item colSpan={12} />
            </Grid.Section>
        </Grid>
    );
    expect(container).toMatchSnapshot();
});

test('Render a Grid with Items having spaces between them', () => {
    const {container} = render(
        <Grid>
            <Grid.Item colSpan={4} spaceAfter={8} />
            <Grid.Item colSpan={2} spaceBefore={10} />
        </Grid>
    );
    expect(container).toMatchSnapshot();
});

test('Render a Grid with class names attached', () => {
    const {container} = render(
        <Grid className="test-grid">
            <Grid.Section className="test-section">
                <Grid.Item className="test-item" />
            </Grid.Section>
        </Grid>
    );
    expect(container).toMatchSnapshot();
});
