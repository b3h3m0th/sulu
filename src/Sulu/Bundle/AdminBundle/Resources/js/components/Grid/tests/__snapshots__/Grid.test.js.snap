// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a Grid with Items having spaces between them 1`] = `undefined`;

exports[`Render a Grid with Items in all sizes 1`] = `undefined`;

exports[`Render a Grid with Sections 1`] = `undefined`;

exports[`Render a Grid with class names attached 1`] = `undefined`;
